torchdata/dataloader2/communication/iter.py

```diff
@@ -263,8 +263,6 @@ def reset(self):
     def reset_epoch(
         self, reset_fn: Callable[[WorkerInfo, SeedGenerator, DataPipe], DataPipe], seed_generator: SeedGenerator
     ):
-        for dp in self.datapipes:
-            dp.protocol.discard_existing_request()
```

Comment:
Is this no longer needed?

Reply:
I mean I can keep it but TBH I don't think it's really useful.